src/types/colorTable.cpp

```diff
@@ -286,85 +286,123 @@ static constexpr std::array<til::color, 256> standardXterm256ColorTable{
     til::color{ 0xEE, 0xEE, 0xEE },
 };
 
+// Routine Description:
+// - Calculate XOrg app color variant with given base color and variant index.
+//   See also https://en.wikipedia.org/wiki/X11_color_names
+// Arguments:
+// - baseColor - The base color to use.
+// - variant - The variant index.
+// Return Value:
+// - The result color
+static constexpr til::color _calculateXorgAppColorVariant(til::color baseColor, size_t variant)
+{
+    float ratio = 1.0f;
+    switch (variant)
+    {
+    case 1:
+        ratio = 1.0f;
+        break;
+    case 2:
+        ratio = 0.932f;
+        break;
+    case 3:
+        ratio = 0.804f;
+        break;
+    case 4:
+        ratio = 0.548f;
+        break;
+    default:
+        break;

```

Comment:
if you switch to a similar algorithm to the one we used for gray ((255*932)+500/1000), you'll get rounding behavior on truncate for free _and_ we won't need floating point math :smile: